cli: use a context struct for the 'cert' command configuration

knz opened the pull request:
Release justification: low risk, high benefit changes to existing functionality

knz committed:
cli: use a context struct for the 'cert' command configuration

Also this commit simplifies the setup of command-line flags
for the 'cert' commands.

Release justification: low risk, high benefit changes to existing functionality

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/69143)
<!-- Reviewable:end -->

knz committed:
cli: avoid an init function in the cert code

Release justification: low risk, high benefit changes to existing functionality

Release note: None

mberhault reviewed (approved):
> Reviewed 4 of 4 files at r2, 5 of 5 files at r3, 30 of 30 files at r4, 2 of 2 files at r5, 38 of 38 files at r6, 4 of 4 files at r7, 1 of 1 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/69143)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @knz)



<!-- Sent from Reviewable.io -->

knz commented:
bors r=otan,mberhault
